cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -74,7 +81,16 @@ func (c *CStorPoolController) syncHandler(key string, operation common.QueueOper
 		return err
 	}
 	klog.V(4).Infof("Lease acquired successfully on csp %s ", cspObject.Name)
-
+	cspGot, err := c.populateVersion(cspObject)
+	if err != nil {
+		klog.Errorf("failed to add versionDetails to cstorpool %s:%s", cspObject.Name, err.Error())
+		return err
+	}
+	cspGot, err = c.reconcileVersion(cspGot)
+	if err != nil {
+		klog.Errorf("failed to upgrade CSP %s:%s", cspObject.Name, err.Error())
+		return err
+	}
```

Comment:
Here, we need to assign cspObject = cspGot we should always deal with the latest object.

Reply:
Done